envoy/tracing/trace_context.h

```diff
@@ -14,15 +15,51 @@ namespace Tracing {
  * Protocol-independent abstraction for traceable stream. It hides the differences between different
  * protocol and provides tracer driver with common methods for obtaining and setting the tracing
  * context.
- *
- * TODO(wbpcode): A new interface should be added to obtain general traceable stream information,
- * such as host, RPC method, protocol identification, etc. At the same time, a new interface needs
- * to be added to support traversal of all trace contexts.
  */
 class TraceContext {
 public:
   virtual ~TraceContext() = default;
 
+  using IterateCallback = std::function<bool(absl::string_view key, absl::string_view val)>;
+
+  /**
+   * Get context protocol.
+   *
+   * @return A string view representing the protocol of the traceable stream behind the context.
+   */
+  virtual absl::string_view contextProtocol() const PURE;
+
+  /**
+   * Get context authority.
+   *
+   * @return The authority of traceable stream. It generally consists of the host and an optional
+   * user information and an optional port.
+   */
+  virtual absl::string_view contextAuthority() const PURE;
+
+  /**
+   * Get context path.
+   *
+   * @return The path of traceable stream. The content and meaning of path are determined by
+   * specific protocol itself.
+   */
+  virtual absl::string_view contextPath() const PURE;
+
+  /**
+   * Get context method.
+   *
+   * @return The method of traceable stream. The content and meaning of method are determined by
+   * specific protocol itself.
+   */
+  virtual absl::string_view contextMethod() const PURE;
+
+  /**
+   * Iterate over all context entry.
+   *
+   * @param callback supplies the iteration callback.
+   */
+  virtual void iterateContext(IterateCallback callback) const PURE;
```

Comment:
It could be good to use &&

Reply:
I am not sure if this is necessary. The current interface can still use move semantics to avoid copying. 🤔